### PR TITLE
refactor: Bump ESLint to 9.6.0

### PR DESCRIPTION
**Pull request by dplewis**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-JS/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-JS/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).

## Issue

Closes: https://github.com/parse-community/Parse-SDK-JS/pull/2202

## Approach

* Replace `dtslint` with `tslint` for linting
* Bump ESLint to 9.6.0
* Handle unused variables
* Improve offlineQuery type checking for $relativeTime
* Improve tests that use `done`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,8 +25,6 @@ jobs:
       - run: npm ci
       - name: Build Types
         run: npm run build:types
-      - name: Test Types
```
Comment (mtrezza) [inline .github/workflows/ci.yml]: Should this be readded?

Comment (dplewis) [reply .github/workflows/ci.yml]: Good catch

**Review by mtrezza — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-alpha.3](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.3.0-alpha.3)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/6.0.0)

**Pull request merged**